packages/components/fields/number-field/src/number-field.tsx

```diff
@@ -0,0 +1,259 @@
+import {
+  Component,
+  isValidElement,
+  type ChangeEventHandler,
+  type FocusEventHandler,
+  type ReactElement,
+  type ReactNode,
+  type MouseEvent,
+  type KeyboardEvent,
+} from 'react';
+import {
+  createSequentialId,
+  filterDataAttributes,
+  getFieldId,
+  warning,
+} from '@commercetools-uikit/utils';
+import Constraints from '@commercetools-uikit/constraints';
+import Stack from '@commercetools-uikit/spacings-stack';
+import FieldLabel from '@commercetools-uikit/field-label';
+import FieldErrors from '@commercetools-uikit/field-errors';
+import NumberInput from '@commercetools-uikit/number-input';
+
+const sequentialId = createSequentialId('number-field-');
+
+type TFieldErrors = Record<string, boolean>;
+type TErrorRenderer = (key: string, error?: boolean) => ReactNode;
+
+const hasErrors = (errors?: TFieldErrors) =>
+  errors && Object.values(errors).some(Boolean);
+
+type TNumberFieldProps = {
+  // NumberField
+  /**
+   * Used as HTML id property. An id is auto-generated when it is not specified.
+   */
+  id?: string;
+  /**
+   * Horizontal size limit of the input fields.
+   */
+  horizontalConstraint?:
+    | 3
+    | 4
+    | 5
+    | 6
+    | 7
+    | 8
+    | 9
+    | 10
+    | 11
+    | 12
+    | 13
+    | 14
+    | 15
+    | 16
+    | 'scale'
+    | 'auto';
+  /**
+   * A map of errors. Error messages for known errors are rendered automatically.
+   * <br />
+   * Unknown errors will be forwarded to `renderError`
+   */
+  errors?: TFieldErrors;
+  /**
+   * Called with custom errors. This function can return a message which will be wrapped in an ErrorMessage. It can also return null to show no error.
+   * <br />
+   * Signature: `(key, error) => React.node`
+   */
+  renderError?: TErrorRenderer;
+  /**
+   * Indicates if the value is required. Shows an the "required asterisk" if so.
+   */
+  isRequired?: boolean;
+  /**
+   * Indicates whether the field was touched. Errors will only be shown when the field was touched.
+   */
+  touched?: boolean;
+
+  // NumberInput
+  /**
+   * Used as HTML name of the input component. property
+   */
+  name?: string;
+  /**
+   * Value of the input component.
+   */
+  value: string | number;
+  /**
+   * Used as HTML `autocomplete` of the input component. property
+   */
+  autoComplete?: string;
+  /**
+   * Called with an event containing the new value. Required when input is not read only. Parent should pass it back as value.
+   * <br />
+   * Signature: `(event) => void`
```

Comment:
You can remove these `Signature` comments. With TS now they are provided by the generator.